src/io/flutter/run/daemon/DeviceService.java

```diff
@@ -230,9 +232,12 @@ private DeviceDaemon chooseNextDaemon(Refreshable.Request<DeviceDaemon> request)
   public void restart() {
     if (project.isDisposed()) return;
 
-    deviceDaemon.refresh(this::shutDownDaemon);
+    JobScheduler.getScheduler().schedule(this::shutDown, 0, TimeUnit.SECONDS);
+    JobScheduler.getScheduler().schedule(this::refreshDeviceDaemon, 4, TimeUnit.SECONDS);
```

Comment:
Ah, or is this the delay for https://github.com/flutter/flutter/issues/8439?  (If so, maybe a TODO to remove?)